pkg/registrar/map_functions.go

```diff
@@ -806,23 +383,14 @@ func (mapper *Mapper) ClusterRoleBindingToRunnableRequests(clusterRoleBindingObj
 		return nil
 	}
 
-	for _, subject := range clusterRoleBinding.Subjects {
-		if subject.APIGroup == "" && subject.Kind == "ServiceAccount" {
-			serviceAccountObject := &corev1.ServiceAccount{}
-			serviceAccountKey := client.ObjectKey{
-				Namespace: subject.Namespace,
-				Name:      subject.Name,
-			}
-			err := mapper.Client.Get(context.TODO(), serviceAccountKey, serviceAccountObject)
-			if err != nil {
-				mapper.Logger.Error(fmt.Errorf("client get: %w", err), "cluster role binding to runnable requests: get service account")
-				return []reconcile.Request{}
-			}
-			return mapper.ServiceAccountToRunnableRequests(serviceAccountObject)
-		}
+	serviceAccounts := mapper.getServiceAccounts(clusterRoleBinding.Subjects)
+
+	var requests []reconcile.Request
+	for _, serviceAccount := range serviceAccounts {
+		requests = append(requests, mapper.ServiceAccountToRunnableRequests(serviceAccount)...)
```

Comment:
not really sure why, but I noticed in the logs that I have some messages from `runnable` logger printing workload mapping-related errors 🤔  

```json
{                                                                                                                                                                                           "level": "error",
  "ts": 1644336477.506691,
  "logger": "runnable",
  "msg": "role binding to workload requests: get service account",
  "error": "client get: ServiceAccount \"route-controller\" not found",
  "stacktrace": "github.com/vmware-tanzu/cartographer/pkg/registrar.(*Mapper).ClusterRoleBindingToRunnableRequests\n\t/cartographer/pkg/registrar/map_functions.go:386\nsigs.k8s.io/      controller-runtime/pkg/handler.(*enqueueRequestsFromMapFunc).mapAndEnqueue..."
}
```

Reply:
fixed in latest commit, thanks!